docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,600 @@
+point in an application `application.ts` is as simple as:
+
+```ts
+registerAuthenticationStrategy(this, BasicAuthenticationStrategy);
+```
+
+## Authentication Decorator
+
+Once you have a **registered** authentication strategy with a **unique** name,
+then you are ready to use the
+[Authentication Decorator](https://loopback.io/doc/en/lb4/Decorators_authenticate.html)
+on your controller endpoints.
+
+They decorator's syntax is :
+
+```ts
+@authenticate(strategyName: string, options?: Object)
+```
+
+The strategy `name` is **mandatory** and the `options` object is **optional**.
+When the options object is specified, it must be relevant to that particular
+strategy.
+
+To **secure** a specific endpoint, you simply need to add the decorator and
+specify the `name` of a registered authentication strategy.
+
+Here is an example of the decorator using the 'basic' authentication strategy we
+defined earlier, **without** options, for the endpoint `/whoami` in a controller
+named `WhoAmIController`.
+
+```ts
+import {inject} from '@loopback/context';
+import {
+  AuthenticationBindings,
+  UserProfile,
+  authenticate,
+} from '@loopback/authentication';
+import {get} from '@loopback/rest';
+
+export class WhoAmIController {
+  constructor(
+    @inject(AuthenticationBindings.CURRENT_USER)
+    private userProfile: UserProfile,
+  ) {}
+
+  @authenticate('basic')
+  @get('/whoami')
+  whoAmI(): string {
+    return this.userProfile.id;
+  }
+}
+```
+
+An example of the decorator when options **are** specified looks like this:
+
+```ts
+@authenticate('basic', { propertyOne : "foo", propertyTwo:"bar"})
+```
+
+{% include tip.html content="
+To avoid repeating the same options in the <b>@authenticate</b> decorator for many endpoints in a controller, you can instead define global options which can be injected into an authentication strategy thereby allowing you to avoid specifying the options object in the decorator itself. For controller endpoints that need to override a global option, you can specify it in an options object passed into the decorator. Your authentication strategy would need to handle the option overrides.
+" %}
+
+After a request is successfully authenticated, the current user profile is
+available on the request context. You can obtain it via dependency injection by
+using the `AuthenticationBindings.CURRENT_USER` binding key.
+
+Parameters of the `@authenticate` decorator can be obtained via dependency
+injection using the `AuthenticationBindings.METADATA` binding key. It returns
+data of type `AuthenticationMetadata` provided by `AuthMetadataProvider`. The
+`AuthenticationStrategyProvider` in the previous section makes use of
+`AuthenticationMetadata` to figure out what **name** you specified as a
+parameter in the `@authenticate` decorator of a specific controller endpoint.
+
+## Adding an Authentication Action to a Sequence
+
+In a LoopBack 4 REST application, each request passes through a stateless
```

Comment:
does it only applicable to LB4 application with REST server? I find it a bit strange to refer it as "LoopBack 4 REST application".

Reply:
@dhmlau , I see what you mean. A user can create a gRPC server I guess, and not a typical REST API server which uses HTTP requests. Are people using LoopBack for more than REST servers at the moment?

Reply:
i think they can, right? @raymondfeng

Reply:
The default sequence mentioned [here](https://github.com/strongloop/loopback-next/pull/2977/files#diff-f95d62f4892a8c070795ee268ae7ca15R335) is for REST server only.

And please note that in the `AuthenticationStrategy` interface, the `Request` type is imported from `@loopback/rest`, see https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/types.ts#L7

So mentioning `REST application` is ok to me here.

Those services like `UserService` and `TokenService` are reusable for other Clients, but the authentication component does target on the REST request.

Reply:
The sequence so far only applies to RestServer. Maybe we can say `In a LoopBack 4 application with REST API endpoints`.